hydra-node/src/Hydra/Chain/Direct/Wallet.hs

```diff
@@ -214,7 +215,7 @@ applyTxs txs isOurs utxo =
       -- XXX: Use cardano-api types instead here
       let tx = toLedgerTx apiTx
       let txId = getTxId tx
-      modify (`Map.withoutKeys` spendInputs' (body tx))
+      modify (`Map.withoutKeys` view spendableInputsTxBodyF (body tx))
```

Comment:
Must: use `inputsTxBodyL` as this is not the same

Looking at `Cardano.Ledger.Babbage.TxBody.html#spendInputs'` and `Cardano.Ledger.Babbage.TxBody.html#babbageSpendableInputsTxBodyF` makes me believe that these are not the same. According to the the docs of `spendableInputsTxBodyF`:
> This getter will produce all inputs from the UTxO map that this transaction might spend, which ones will depend on the validity of the transaction itself. Starting in Alonzo this will include collateral inputs.

Reply:
Shouldn't this break a test somewhere?

Reply:
No because we re-use an input as a collateral. So for us, the union would just collapse. 

I hope it's okay that I pointed this out still :)